samples/sensor/stm32_temp_sensor/sample.yaml

```diff
@@ -5,5 +5,10 @@ tests:
   sample.sensor.stm32_temp_sensor:
     depends_on: adc
     tags: sensors
-    build_only: true
-    platform_allow: nucleo_f401re stm32f103_mini stm32l562e_dk
+    tags: tests
+    filter: dt_compat_enabled("st,stm32-temp-cal") or dt_compat_enabled("st,stm32-temp")
+    harness: console
+    harness_config:
+      type: one_line
+      regex:
+        - r"Current temperature ^([0-9]{1,2}){1}(\.[0-9]{1,2}) °C"
```

Comment:
Modifications on sample part should be part of a separate commit.

I think that Regex could be simplified:
r"Current temperature [0-9]{1,2}\.[0-9]{1,2} °C"

Reply:
+1 for the separate commit.
Regarding regex, let's limit expected t° range to 10>60.
Question on the "floating" part: Is it possible to get 2 digit ?

Reply:
According the example, the float part is composed of a single digit.